test/lib/jdk/test/lib/thread/ProcessThread.java

```diff
@@ -150,16 +150,23 @@ public ProcessRunnable(ProcessBuilder pb, String name, Predicate<String> waitfor
          */
         @Override
         public void xrun() throws Throwable {
-            this.process = ProcessTools.startProcess(name, processBuilder, waitfor);
-            // Release when process is started
-            latch.countDown();
+            String name = Thread.currentThread().getName();
+
+            try {
+                this.process = ProcessTools.startProcess(name, processBuilder, waitfor);
+            } catch (Throwable t) {
+                System.out.println(String.format("ProcessThread[%s] failed: %s", name, t.toString()));
+                throw t;
+            } finally {
+                // Release when process is started or failed
+                latch.countDown();
+            }
 
-            // Will block...
             try {
-                this.process.waitFor();
                 output = new OutputAnalyzer(this.process);
+                // Will block...
+                this.process.waitFor();
```

Comment:
It would be useful to capture the exit status here for the log.

Reply:
It is printed anyway in ProcessTools.getProcessLog(...)